src/lib/backend_fapi.c

```diff
@@ -68,6 +113,14 @@ static char * tss_path_from_id(unsigned id, const char *type) {
     return path;
 }
 
+static char *path_get_parent(const char *path) {
+    char *end = rindex(path, '/');
+    if (!end) {
+        return NULL;
+    }
+    return strndup(path, end - path);
+}
```

Comment:
I'm going to file a bug, but I am concerned with all the path manipulation string parsing. So we need to make sure that code is robust, can we get some fuzz tests? Not necessarily now, but in the future.

Reply:
We're actually just tokenizing stuff on '\0' whilst keeping buffer-checks.
The rest is sscanf("%08x") and the yaml-parser.
So yes, looks scary, but with simple operations.

Fuzzing is however never a bad idea...